components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/store/SessionDataStore.java

```diff
@@ -330,28 +331,29 @@ public void removeExpiredSessionData(Timestamp timestamp) {
             log.error(e.getMessage(), e);
             return;
         }
+        long cleanupLimit = TimeUnit.MINUTES.toNanos(FrameworkUtils.getCurrentStandardNano() - IdentityUtil
+                .getCleanUpTimeout());
         try {
             statement = connection.prepareStatement(sqlDeleteExpiredDataTask);
-            long currentStandardNano = getCurrentStandardNano(timestamp);
-            statement.setLong(1, currentStandardNano);
+            statement.setLong(1, cleanupLimit);
             statement.execute();
             if (!connection.getAutoCommit()) {
                 connection.commit();
             }
         } catch (SQLException e) {
-            log.error("Error while removing session data from the database for the timestamp " + timestamp.toString(), e);
+            log.error("Error while removing session data from the database for nano time " + cleanupLimit, e);
         } finally {
             IdentityDatabaseUtil.closeAllConnections(connection, null, statement);
 
         }
     }
 
-    public void removeExpiredOperationData(Timestamp timestamp) {
-        deleteSTOREOperationsTask(timestamp);
-        deleteDELETEOperationsTask(timestamp);
+    public void removeExpiredOperationData() {
+        deleteSTOREOperationsTask();
+        deleteDELETEOperationsTask();
     }
 
-    public void persistSessionData(String key, String type, Object entry, Timestamp timestamp, int tenantId) {
+    public void persistSessionData(String key, String type, Object entry, long nanoTime, int tenantId) {
```

Comment:
This method is consumed by other components (like SAML) as well. Hence it would need to refactor all usages with this change.